From 1f37cc5626232e8636729c8ee620c27da91aa39b Mon Sep 17 00:00:00 2001
Message-ID: <1f37cc5626232e8636729c8ee620c27da91aa39b.1686001816.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Mon, 5 Jun 2023 14:00:23 +0200
Subject: [PATCH] [rest] initialize dataset

When there is already a dataset, and setting properties using the PUT
method and an incomplete JSON dataset, some fields of the dataset
variable might not be initialized. This leads to uninitialized data
being used as active dataset.

Make sure we initialize the dataset variable. This makes sure that all
components of the dataset are considered not set.
---
 src/rest/resource.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rest/resource.cpp b/src/rest/resource.cpp
index a92754a381..0327ee2996 100644
--- a/src/rest/resource.cpp
+++ b/src/rest/resource.cpp
@@ -659,7 +659,7 @@ void Resource::SetDataset(DatasetType aDatasetType, const Request &aRequest, Res
     struct NodeInfo          node;
     std::string              body;
     std::string              errorCode = GetHttpStatus(HttpStatusCode::kStatusOk);
-    otOperationalDataset     dataset;
+    otOperationalDataset     dataset   = {};
     otOperationalDatasetTlvs datasetTlvs;
     otOperationalDatasetTlvs datasetUpdateTlvs;
     int                      ret;
-- 
2.41.0

